entrypoint.sh

```diff
@@ -1,59 +0,0 @@
-#!/bin/bash
-
-: ${SSH_KEY:=$HOME/ssh-private-rsa.key}
-
-# When mounting a volume; docker ( <= v1.10.3) does not provide means to set permissions on the mount point
-# "setup-workspace" subcommand is a workaround for that
-if [[ "$1" != setup-workspace && ( \
-    ! -w $HOME || \
-    ! -d $HOME/.ssh || \
-    ( -f "${SSH_KEY}" && ! -f $HOME/.ssh/id_rsa )
-) ]]; then
-    # we need root privileges to change permission of the directory
-    if ! sudo "$0" setup-workspace; then
-        echo setup-workspace failed\; giving up.;
-        exit 1;
-    fi
-fi
-
-case "$1" in
-    setup-workspace)
-        if [ ! -w $HOME ]; then
-            # change permission; not recursive by purpose
-            # recursive chown can take ages if the volume has a lot of files and directories
-            exec chown migrator:migrator $HOME;
-        fi
-
-        if [ ! -d $HOME/.ssh ]; then
-            mkdir $HOME/.ssh &&
-                chmod go-rwx $HOME/.ssh &&
-                chown migrator:migrator $HOME/.ssh;
-        fi
-
-        if [[ -f "${SSH_KEY}" && ! -f $HOME/.ssh/id_rsa ]]; then
-            cp "${SSH_KEY}" $HOME/.ssh/id_rsa &&
-                chown migrator:migrator $HOME/.ssh/id_rsa &&
-                chmod go-rwx $HOME/.ssh/id_rsa;
-        fi
-        ;;
-
-    wait-up)
```

Comment:
I think the Docker-y answer to the question of how to run long running migrations which won't fail if the user disconnects is to detach from the container.  Once you've kicked off the `svn2git` command you can detach with `Ctrl-P + Ctrl-Q`.

Alternatively, you could kick off the migration in detached mode without ever entering the container.  For example, the following few commands get the list of authors, kick off the `svn2git` migration in detached mode then use docker logs to watch the output:
```
docker run --rm --network=host -v ~/.ssh/known_hosts:/root/.ssh/known_hosts -v ${SSH_AUTH_SOCK}:${SSH_AUTH_SOCK} -v ${PWD}/workspace:/workspace -e SSH_AUTH_SOCK=${SSH_AUTH_SOCK} redmatter/svn2git-workspace sh -c "get-authors svn+ssh://user@repo/trunk email.com > authors.txt"
docker run -d --name=my-long-running-git-migration --network=host -v ~/.ssh/known_hosts:/root/.ssh/known_hosts -v ${SSH_AUTH_SOCK}:${SSH_AUTH_SOCK} -v ${PWD}/workspace:/workspace -e SSH_AUTH_SOCK=${SSH_AUTH_SOCK} redmatter/svn2git-workspace sh -c "svn2git svn+ssh://user@repo/trunk --svn2git-args"
docker logs -f my-long-running-git-migration
```
While a lot of that would be needed anyway (the volumes and environment variables), it is a bit long-winded, so potentially could be pulled into the `docker-compose.yml` if we wanted to support this mechanism.  I think simply detaching from the container if you need to is probably the best approach, though.

I don't want to completely change the image if you're not happy with it, so if you really think we need to support `screen` etc I can try to re-instate these files, but I removed them in the first place because the home directory set up in `entrypoint.sh` was interfering with my ability to mount the SSH agent in, so at the time if felt like it was more trouble than it was worth to try and maintain.